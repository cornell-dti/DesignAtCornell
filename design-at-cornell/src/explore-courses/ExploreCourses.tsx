import React, { useState } from 'react';
import Title from './title/Title';
import {
  PageContainer,
  Sort,
  CourseGrid,
  CourseBubble,
  TagsContainer,
  Tag
} from './ExploreCoursesStyles';

type Course = {
  courseId: string;
  courseTitle: string;
  tags: string[];
};

const ExploreCourses = () => {
  const [courses, setCourses] = useState<Course[]>([
    {courseId: "INFO 1300", courseTitle: "Introductory Design and Programming for the Web", tags: ["Information Science", "3 Credits"]},
    {courseId: "DEA 1500", courseTitle: "Introduction to Environmental Psychology", tags: ["Design and Enviro...", "3 Credits"]},
    {courseId: "FSAD 1120", courseTitle: "Fashion Design and Visual Thinking", tags: ["Fashion Design an...", "Fall"]},
    {courseId: "ART 2201", courseTitle: "Painting: Introduction to Painting", tags: ["Fine Arts", "3 Credits"]},
    {courseId: "ARCH 1101", courseTitle: "Design I", tags: ["Architecture", "3 Credits", "Fall"]},
    {courseId: "ART 2301", courseTitle: "Print Media: Introduction to Print Media", tags: ["Fine Arts", "3 Credits"]},
  ]);

  const courseBubbles = (
    <CourseGrid>
      {courses.map(course => (
        <CourseBubble>
          <h6>{course.courseId}</h6>
          <p>{course.courseTitle}</p>
          <TagsContainer>
            {course.tags.map(tag => (
              <Tag>
                <p>{tag}</p>
              </Tag>
            ))}
          </TagsContainer>
        </CourseBubble>
      ))}
    </CourseGrid>
  );

  return (
    <PageContainer>
    <Title />
    <Sort>
      <p>Sort By</p>
    </Sort>
    {courseBubbles}
  </PageContainer>
  );
};

export default ExploreCourses;