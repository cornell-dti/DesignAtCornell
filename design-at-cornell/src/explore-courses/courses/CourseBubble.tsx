import React from 'react';
import {
  ElementContainer,
  Subtitle,
  TagsContainer,
  Tag,
} from '../../components/DashboardElementStyles';
import { courseColors } from '../../constants/colors';
import bookmark from '../../static/images/bookmark.svg';
import { Course } from '../../../../server/types';

const CourseBubble = (course: Course) => (
  <ElementContainer
    style={{ borderColor: courseColors[Math.floor(course.code % courseColors.length)] }}
    onClick={() => (window.location.href = course.content.courseRoster)}
  >
    <Subtitle>
      <p>{course.id + ' ' + course.code}</p>
      <img src={bookmark} alt="save course" />
    </Subtitle>
    <p>{course.content.title}</p>
    <TagsContainer>
      <Tag style={{ background: courseColors[0] }}>
        <p>{course.content.credits + ' Credits'}</p>
      </Tag>
      <Tag style={{ background: courseColors[1] }}>
        <p>{course.content.semester.join(', ')}</p>
      </Tag>
    </TagsContainer>
  </ElementContainer>
);

export default CourseBubble;
