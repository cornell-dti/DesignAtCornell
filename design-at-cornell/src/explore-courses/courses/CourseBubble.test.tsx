import React from 'react';
import renderer from 'react-test-renderer';
import CourseBubble from './CourseBubble';
import { Course } from '../ExploreCourses';

describe('<CourseBubble />', () => {
  it('renders testing course correctly', () => {
    const course : Course = {
      courseCode: 'TEST 1001',
      courseTitle: 'Testing Course Title',
      tags: ['tag 1', 'tag 2', 'tag 3']
    }

    const tree = renderer
      .create(
        <CourseBubble 
          courseCode={course.courseCode}
          courseTitle={course.courseTitle}
          tags={course.tags}  
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('reders course with empty course code', () => {
    const course : Course = {
      courseCode: '',
      courseTitle: 'Testing Course Title',
      tags: ['tag 1', 'tag 2', 'tag 3']
    }

    const tree = renderer
      .create(
        <CourseBubble 
          courseCode={course.courseCode}
          courseTitle={course.courseTitle}
          tags={course.tags}  
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders course with empty course title', () => {
    const course : Course = {
      courseCode: 'TEST 1002',
      courseTitle: '',
      tags: ['tag 1', 'tag 2', 'tag 3']
    }

    const tree = renderer
      .create(
        <CourseBubble 
          courseCode={course.courseCode}
          courseTitle={course.courseTitle}
          tags={course.tags}  
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders course with empty tags list', () => {
    const course : Course = {
      courseCode: 'TEST 1002',
      courseTitle: '',
      tags: []
    }

    const tree = renderer
      .create(
        <CourseBubble 
          courseCode={course.courseCode}
          courseTitle={course.courseTitle}
          tags={course.tags}  
      />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
