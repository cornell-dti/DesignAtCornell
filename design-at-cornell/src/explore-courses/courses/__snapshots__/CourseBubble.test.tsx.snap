// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CourseBubble /> renders course with empty course title 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 22%;
  min-width: 250px;
  height: 197px;
  cursor: pointer;
  background: white;
  border: 1px solid;
  border-radius: 8px;
  margin: 1.5%;
  padding: 25px;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 p {
  font-weight: bold;
  font-size: 20px;
  line-height: 23px;
  text-align: start;
  margin: 0;
  color: black;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 p {
  font-weight: bold;
  font-size: 16px;
  line-height: 19px;
  text-align: start;
  margin: 0;
  color: #898989;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  width: 100%;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: 28px;
  border-radius: 43px;
  margin-right: 10px;
  padding-left: 10px;
  padding-right: 10px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 p {
  white-space: nowrap;
  font-weight: bold;
  font-size: 12px;
  color: black;
}

<div
  className="c0"
  onClick={[Function]}
  style={
    Object {
      "borderColor": "#C9E1AA",
    }
  }
>
  <div
    className="c1"
  >
    <p>
      TEST 1101
    </p>
    <img
      alt="save course"
      src="bookmark.svg"
    />
  </div>
  <p>
    
  </p>
  <div
    className="c2"
  >
    <div
      className="c3"
      style={
        Object {
          "background": "#BAE6FF",
        }
      }
    >
      <p>
        4 Credits
      </p>
    </div>
    <div
      className="c3"
      style={
        Object {
          "background": "#FFE5A1",
        }
      }
    >
      <p>
        Fall, Spring
      </p>
    </div>
  </div>
</div>
`;

exports[`<CourseBubble /> renders course with empty description 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 22%;
  min-width: 250px;
  height: 197px;
  cursor: pointer;
  background: white;
  border: 1px solid;
  border-radius: 8px;
  margin: 1.5%;
  padding: 25px;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 p {
  font-weight: bold;
  font-size: 20px;
  line-height: 23px;
  text-align: start;
  margin: 0;
  color: black;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 p {
  font-weight: bold;
  font-size: 16px;
  line-height: 19px;
  text-align: start;
  margin: 0;
  color: #898989;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  width: 100%;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: 28px;
  border-radius: 43px;
  margin-right: 10px;
  padding-left: 10px;
  padding-right: 10px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 p {
  white-space: nowrap;
  font-weight: bold;
  font-size: 12px;
  color: black;
}

<div
  className="c0"
  onClick={[Function]}
  style={
    Object {
      "borderColor": "#C9E1AA",
    }
  }
>
  <div
    className="c1"
  >
    <p>
      TEST 1101
    </p>
    <img
      alt="save course"
      src="bookmark.svg"
    />
  </div>
  <p>
    Intro to Design
  </p>
  <div
    className="c2"
  >
    <div
      className="c3"
      style={
        Object {
          "background": "#BAE6FF",
        }
      }
    >
      <p>
        4 Credits
      </p>
    </div>
    <div
      className="c3"
      style={
        Object {
          "background": "#FFE5A1",
        }
      }
    >
      <p>
        Fall, Spring
      </p>
    </div>
  </div>
</div>
`;

exports[`<CourseBubble /> renders course with empty designAreas list 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 22%;
  min-width: 250px;
  height: 197px;
  cursor: pointer;
  background: white;
  border: 1px solid;
  border-radius: 8px;
  margin: 1.5%;
  padding: 25px;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 p {
  font-weight: bold;
  font-size: 20px;
  line-height: 23px;
  text-align: start;
  margin: 0;
  color: black;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 p {
  font-weight: bold;
  font-size: 16px;
  line-height: 19px;
  text-align: start;
  margin: 0;
  color: #898989;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  width: 100%;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: 28px;
  border-radius: 43px;
  margin-right: 10px;
  padding-left: 10px;
  padding-right: 10px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 p {
  white-space: nowrap;
  font-weight: bold;
  font-size: 12px;
  color: black;
}

<div
  className="c0"
  onClick={[Function]}
  style={
    Object {
      "borderColor": "#C9E1AA",
    }
  }
>
  <div
    className="c1"
  >
    <p>
      TEST 1101
    </p>
    <img
      alt="save course"
      src="bookmark.svg"
    />
  </div>
  <p>
    Intro to Design
  </p>
  <div
    className="c2"
  >
    <div
      className="c3"
      style={
        Object {
          "background": "#BAE6FF",
        }
      }
    >
      <p>
        4 Credits
      </p>
    </div>
    <div
      className="c3"
      style={
        Object {
          "background": "#FFE5A1",
        }
      }
    >
      <p>
        Fall, Spring
      </p>
    </div>
  </div>
</div>
`;

exports[`<CourseBubble /> renders course with empty major 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 22%;
  min-width: 250px;
  height: 197px;
  cursor: pointer;
  background: white;
  border: 1px solid;
  border-radius: 8px;
  margin: 1.5%;
  padding: 25px;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 p {
  font-weight: bold;
  font-size: 20px;
  line-height: 23px;
  text-align: start;
  margin: 0;
  color: black;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 p {
  font-weight: bold;
  font-size: 16px;
  line-height: 19px;
  text-align: start;
  margin: 0;
  color: #898989;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  width: 100%;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: 28px;
  border-radius: 43px;
  margin-right: 10px;
  padding-left: 10px;
  padding-right: 10px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 p {
  white-space: nowrap;
  font-weight: bold;
  font-size: 12px;
  color: black;
}

<div
  className="c0"
  onClick={[Function]}
  style={
    Object {
      "borderColor": "#C9E1AA",
    }
  }
>
  <div
    className="c1"
  >
    <p>
      TEST 1101
    </p>
    <img
      alt="save course"
      src="bookmark.svg"
    />
  </div>
  <p>
    Intro to Design
  </p>
  <div
    className="c2"
  >
    <div
      className="c3"
      style={
        Object {
          "background": "#BAE6FF",
        }
      }
    >
      <p>
        4 Credits
      </p>
    </div>
    <div
      className="c3"
      style={
        Object {
          "background": "#FFE5A1",
        }
      }
    >
      <p>
        Fall, Spring
      </p>
    </div>
  </div>
</div>
`;

exports[`<CourseBubble /> renders course without roster 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 22%;
  min-width: 250px;
  height: 197px;
  cursor: pointer;
  background: white;
  border: 1px solid;
  border-radius: 8px;
  margin: 1.5%;
  padding: 25px;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 p {
  font-weight: bold;
  font-size: 20px;
  line-height: 23px;
  text-align: start;
  margin: 0;
  color: black;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 p {
  font-weight: bold;
  font-size: 16px;
  line-height: 19px;
  text-align: start;
  margin: 0;
  color: #898989;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  width: 100%;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: 28px;
  border-radius: 43px;
  margin-right: 10px;
  padding-left: 10px;
  padding-right: 10px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 p {
  white-space: nowrap;
  font-weight: bold;
  font-size: 12px;
  color: black;
}

<div
  className="c0"
  onClick={[Function]}
  style={
    Object {
      "borderColor": "#C9E1AA",
    }
  }
>
  <div
    className="c1"
  >
    <p>
      TEST 1101
    </p>
    <img
      alt="save course"
      src="bookmark.svg"
    />
  </div>
  <p>
    Intro to Design
  </p>
  <div
    className="c2"
  >
    <div
      className="c3"
      style={
        Object {
          "background": "#BAE6FF",
        }
      }
    >
      <p>
        4 Credits
      </p>
    </div>
    <div
      className="c3"
      style={
        Object {
          "background": "#FFE5A1",
        }
      }
    >
      <p>
        Fall, Spring
      </p>
    </div>
  </div>
</div>
`;

exports[`<CourseBubble /> renders testing course correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 22%;
  min-width: 250px;
  height: 197px;
  cursor: pointer;
  background: white;
  border: 1px solid;
  border-radius: 8px;
  margin: 1.5%;
  padding: 25px;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c0 p {
  font-weight: bold;
  font-size: 20px;
  line-height: 23px;
  text-align: start;
  margin: 0;
  color: black;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c1 p {
  font-weight: bold;
  font-size: 16px;
  line-height: 19px;
  text-align: start;
  margin: 0;
  color: #898989;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  width: 100%;
  height: -webkit-fit-content;
  height: -moz-fit-content;
  height: fit-content;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  height: 28px;
  border-radius: 43px;
  margin-right: 10px;
  padding-left: 10px;
  padding-right: 10px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 p {
  white-space: nowrap;
  font-weight: bold;
  font-size: 12px;
  color: black;
}

<div
  className="c0"
  onClick={[Function]}
  style={
    Object {
      "borderColor": "#C9E1AA",
    }
  }
>
  <div
    className="c1"
  >
    <p>
      TEST 1101
    </p>
    <img
      alt="save course"
      src="bookmark.svg"
    />
  </div>
  <p>
    Intro to Design
  </p>
  <div
    className="c2"
  >
    <div
      className="c3"
      style={
        Object {
          "background": "#BAE6FF",
        }
      }
    >
      <p>
        4 Credits
      </p>
    </div>
    <div
      className="c3"
      style={
        Object {
          "background": "#FFE5A1",
        }
      }
    >
      <p>
        Fall, Spring
      </p>
    </div>
  </div>
</div>
`;
